### PR TITLE
HTML components for Jupyter & RMarkdown

### PR DESCRIPTION
**Pull request by m-mohr**
Open questions:
- [x] Jobs/Services/Files output both a R "table" and a Vue table. Should we just use one? Do both variants display in all environments?
- [x] Service Types/File Formats/UDF Runtimes have issues with the additional ID props: How to solve?
- [x] File Format has issues with the additional type props: How to solve?
- [x] Processes/Service Types/File Formats/UDPs currently don't show the list but one component per entry
- [x] Jobs/Services: Tables are empty?
- [x] Job/Service: Process graph is missing
- [x] UDFs don't work properly -> issue in Vue Components, will resolve once v2.7.1 becomes available through CDN
- [x] Process Graphs (e.g. from Process Builder) need to be checked
- [x] Logs need to be checked
- [x] Currency is not passed - ignore? or what's the best solution? @flahn 
- [x] knitr sometimes works and sometimes doesn't work. why? (due to `return(invisible())` maybe?)
- [x] Job Estimates need to be checked at EODC or mundialis @m-mohr 
- [x] logs() shows no entries: log_job/log_service works, but logs is problematic - see also #109 regarding issues with the timeout
- [x] Check tibble support in Jupyter: Tibble support is given, but doesn't look as nice as in Jupyter so fall back to HTML tables
- [x] Edit icon in the model builder doesn't work

### CODE REVIEW

**Comment by m-mohr** [comment]
Note: For umlauts etc. the following code is required for the Vue components, otherwise they are not rendered correctly:
`options(encoding = "UTF-8")` - How should we handle this?

**Comment by m-mohr** [comment]
Disabled knitr for now, it is pretty much unpredictable in which cases it works and in which cases it does not work.

**Comment by m-mohr** [comment]
This is ready for review @flahn

File: R/print-functions.R [hunk 1]
```diff
@@ -434,21 +505,35 @@ print.OpenEOCapabilities = function(x, ...) {
 
 #' @export
 print.ResultList = function(x, ...) {
-    cat("Results for job: ",x$id,"\n")
-    if (length(x$properties$expires) > 0) {
-        cat("Links expire on: ",x$properties$expires,"\n")
+    if (is_html_context()) {
+        unclass(x$assets)
+        return(print_html("batch-job-results", x))
```
Comment (flahn) [inline R/print-functions.R:510]: I tried to fix this, but I couldn't get it to work. Maybe there is something wrong with the template call. At least I couldn't visualize this. Do need to visualize the assets or the whole job results?

File: R/viewer.R [hunk 2]
```diff
@@ -1,4 +1,80 @@
-htmlViewer = function(html) {
+COMPONENT_MAP = list(
+    'collection' = 'data',
+    'data-table' = 'data',
+    'file-format' = 'format',
+    'file-formats' = 'formats',
+    'item' = 'data',
```
Comment (flahn) [inline R/viewer.R:6]: Maybe here is somethint amiss for `list_results()`

**Review by flahn — COMMENTED** [review]
In general, this looks awesome! Good work! I have some problems to render the HTML for `list_results()` and made some notes about that. Also I wonder if there are vue components for the user account?

File: R/print-functions.R (hunk 1 above)
Comment (m-mohr) [reply R/print-functions.R:510]: I'll check list_results again, but I think it worked flawlessly for me. What issues are you encountering and at which back-end?

File: R/viewer.R (hunk 2 above)
Comment (m-mohr) [reply R/viewer.R:6]: No, should be complete. It works as it is in the Python client.

**Comment by m-mohr** [comment]
No, there's no component (yet) for the user details.

**Pull request merged**